cpp/include/cuspatial/detail/utility/linestring.cuh

```diff
@@ -16,12 +16,19 @@
 
 #pragma once
 
-#include <thrust/tuple.h>
-
 #include <cuspatial/vec_2d.hpp>
 
+#include <thrust/optional.h>
+#include <thrust/pair.h>
+#include <thrust/swap.h>
+#include <thrust/tuple.h>
+
 namespace cuspatial {
 namespace detail {
+
+template <typename T>
+using segment = thrust::pair<vec_2d<T>, vec_2d<T>>;
```

Comment:
I'm leaning toward to creating a separate type for `segment`. A pair of point isn't really a segment - a segment is an **unordered** pair of two points. Also it doesn't work well with initializer_lists.

Reply:
Actually, nevermind. Not really a big concern atm